### PR TITLE
LINK-359 | Filter own event by event type and text

### PR DESCRIPTION
**Pull request by jorilindell**
## Description 
Filter own event by event type and text
Always return to page from where user came from when clicking back button on edit event page
Store sort and page to url query instead of using redux store
Upgrade react to the latest version

## Closes
[LINK-359](https://helsinkisolutionoffice.atlassian.net/browse/LINK-359)

## Screenshot
Screenshot 2021-05-10 at 11 34 07

### CODE REVIEW

**Review by Sartonon — COMMENTED** [review]
Looking very good. Didn't have time to test extensively mut commented some small things.

File: src/domain/auth/silentCallback/SilentCallback.tsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { processSilentRenew } from 'redux-oidc';
 
-export default () => {
+const SilentCallback = (): null => {
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/auth/silentCallback/SilentCallback.tsx:3]: Returning void seems to cause `Type '() => void' is not assignable to type 'FunctionComponent<RouteComponentProps<any, StaticContext, unknown>>'.` error

File: src/common/components/formikPersist/FormikPersist.tsx [hunk 2]
```diff
@@ -14,25 +14,33 @@ const FormikPersist = ({
   debounceTime = 300,
   isSessionStorage = false,
   name,
-}: PersistProps) => {
+}: PersistProps): null => {
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/common/components/formikPersist/FormikPersist.tsx:17]: Oh yes that might be the case

File: src/domain/auth/silentCallback/SilentCallback.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/auth/silentCallback/SilentCallback.tsx:3]: Oh true :)

File: src/domain/events/eventList/EventList.tsx [hunk 3]
```diff
@@ -26,78 +37,147 @@ import EventCard from '../eventCard/EventCard';
...
   skip?: boolean;
-  sort: EVENT_SORT_OPTIONS;
 }
 
 const getPageCount = (count: number, pageSize: number) => {
   return Math.ceil(count / pageSize);
 };
 
+type EventListProps = {
+  events: EventFieldsFragment[];
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/events/eventList/EventList.tsx]: Changed

File: src/domain/event/hooks/useEventFieldOptionsData.ts [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
   useKeywordSetQuery,
   useLanguagesQuery,
 } from '../../../generated/graphql';
 import getPathBuilder from '../../../utils/getPathBuilder';
 import { keywordSetPathBuilder } from '../../keywordSet/utils';
 
+type EventFieldOptionsDataState = {
+  audienceData: KeywordSetQuery | undefined;
+  loading: boolean;
+  topicsData: KeywordSetQuery | undefined;
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/event/hooks/useEventFieldOptionsData.ts]: Of course. Changed

**Pull request merged**